### DISCUSSION TITLE
Pops and cracks when using automation

### DISCUSSION THREAD

**Discussion by shittymodernart** [apohl79/audiogridder#1100] [Help]
I noticed this a few weeks ago, but previously I hardly ever used automation with the plugins I use through Audiogridder so I didn't really look into it. Today, though, I moved some of my instruments over to the AG server and needed to automate the sustain pedal for a piano. Whenever the automation changes the state of the plugin (so anytime automation is used to change anything), the audio pops and crackles.

In order to make this easier to troubleshoot, I created a new session in Logic with a single instrument track with the AG plugin on it and automated the sustain pedal so the logs would be a bit cleaner (see attached). I also bounced the piano part so you can hear what I'm talking about but it doesn't seem like I'm able to attach it to this post. I did a search for "automation" but didn't see anything that sounded like my issue; let me know if I can provide anything else that would be helpful. I also have the plugin and server trace files, but it wouldn't let me upload those here either.

[AudioGridderServer_2022-11-26_18-06-45.log](https://github.com/apohl79/audiogridder/files/10097193/AudioGridderServer_2022-11-26_18-06-45.log)

[AudioGridderPlugin_2022-11-26_18-07-05.log](https://github.com/apohl79/audiogridder/files/10097195/AudioGridderPlugin_2022-11-26_18-07-05.log)

Edit: Should probably include that my DAW computer is an M1 Mac Mini with OSX Monterey 12.4 installed and I'm using Logic Pro X 10.7.5. My server computer is a Windows 10 machine connected directly via ethernet cable. Both plugin and server versions of AG are 1.2.0-beta15.

Edit 2: I threw up the bounce of the crackly piano on my pCloud [here](http://u.pc.cd/z497)

FINAL EDIT: I think I figured it out, at least for now. I had all my AG plugins set at the lowest buffer setting, but when I disabled the buffer it stopped crackling. I need the buffer for other plugins so I just enabled "Allow individual buffer size by plugin" and disabled the buffer for the instrument and everything sounds great! 

This is the first time I'm using AG for an instrument - how common is it to need a buffer for an instrument? And is there a way to have a buffer, use automation, and not have crackles (for future knowledge)?

Thanks so much for all the work you put into this, sorry for the wild ride here lol

**Comment by apohl79** [comment]
Thx for the logs, will take a look!

**Reply by apohl79** [reply]
Thanks for the very clean data! Maybe it helps to find that buffering issue that seems to be still in the code. Have you tried to turn Fixed buffers on and off in AG? Does that change anything?

**Reply by shittymodernart** [reply]
I did read in a comment a while ago that you said the fixed buffer issue was fixed, but whenever I've tried to use it it seems to work fine for a bit then start glitching again. To be honest, I'm not really even sure *why* I would use fixed buffers or what that would solve, so I just turned it off and don't really mess with it.

But mannnnn am I having a terrible time with it tonight; part of the problem is that I have this huge project with over 200 tracks, and a ton of AG plugins active. Looking at the monitor, I don't see any RdErrs running at a 2-block 4096-sample buffer, but all of the audio will (frequently and randomly) become garbled and glitchy. Not using fixed outbound buffers. The only way I'm able to make the glitches go away is to restart the AG server app, but sometimes that doesn't even work and I have completely close out of the daw, restart the server, and go back in. 

Anyway, I didn't come here to make a post about it; I was just tryna see if anyone else was experiencing similar issues or if there was a new beta update released. I don't see the point in making a post because this project is such a mess - I honestly couldn't expect anyone to make sense of it enough to troubleshoot it. Thanks for all you do!

_Edit: Once again, I'm the idiot here. I guess I changed my DAW buffer from 1024 to 128 samples yesterday when I was tracking something and never changed it back. Apparently Logic isn't havin a good time trying to handle 200+ tracks with a 128 sample buffer, who would have guessed?_

**Comment by gatd4** [comment]
same here any buffer over disabled cause glitching and popping when using automation

**Comment by sloofin** [comment]
This may be a niche use case but I have the same problem on Monterey using Parallels and windows for the server. As soon as any automation occurs everything crackles and is unplayable - this happens in Logic, Live, and Bitwig at all buffer settings even with an empty project and a single audiogridder.. But if I make windows full screen and bring it to the front, suddenly everything works fine. It's not a solution, as I can't see the DAW then, but it will at least stop the crackling and allow things to play.

**Comment by LukeG72** [comment]
Hi, same issue here. Logic 10.5.1 on Mojave 10.14.6. Automation glitch with sample overload message.
If I disable automation track, glitches disappear. Buffer setting does not solve the problem.
Is there a beta that fixes this bug?

**Reply by Emmons1983** [reply]
> Hi, same issue here. Logic 10.5.1 on Mojave 10.14.6. Automation glitch with sample overload message.
> If I disable automation track, glitches disappear. Buffer setting does not solve the problem.
> Is there a beta that fixes this bug?

I wish i could help, but i’m no longer experiencing this issue. I’m using Logic Pro 10.7.7 on Ventura 13.3, using audiogridder server and plugin version 1.2. I usually use the lowest buffer or second lowest buffer and haven’t had automation issues in a while (I still hold my breath every time i try it though)

**Reply by LukeG72** [reply]
Thanks, that's a weird issue...
